src/Kunstmaan/NodeSearchBundle/Configuration/NodePagesConfiguration.php

```diff
@@ -577,33 +568,17 @@ protected function enterRequestScope($lang)
      * @param EngineInterface             $renderer
      *
      * @return string
+     *
+     * @deprecated since 5.1. Use "Kunstmaan\NodeSearchBundle\Helper\SearchRenderHelper::renderCustomSearchView" instead.
      */
     protected function renderCustomSearchView(
         NodeTranslation $nodeTranslation,
         SearchViewTemplateInterface $page,
         EngineInterface $renderer
     ) {
-        $view = $page->getSearchView();
-        $renderContext = new RenderContext([
-            'locale'          => $nodeTranslation->getLang(),
-            'page'            => $page,
-            'indexMode'       => true,
-            'nodetranslation' => $nodeTranslation,
-        ]);
-
-        if ($page instanceof PageInterface) {
-            $request = $this->container->get('request_stack')->getCurrentRequest();
-            $page->service($this->container, $request, $renderContext);
-        }
-
-        $content = $this->removeHtml(
-            $renderer->render(
-                $view,
-                $renderContext->getArrayCopy()
-            )
-        );
+        @trigger_error(sprintf('The "%s" method is deprecated in NodeSearchBundle 5.1 and will be removed in NodeSearchBundle 6.0.', __METHOD__), E_USER_DEPRECATED);
```

Comment:
Add also the alternative to the deprecation message 
```
'XXXX deprecated. Use `Kunstmaan\NodeSearchBundle\Helper\SearchRenderHelper::renderCustomSearchView` instead.
``` 

So the user knows how to fix it